src/olympia/addons/tests/test_models.py

```diff
@@ -1456,6 +1456,57 @@ def test_unlisted_has_complete_metadata(self):
         assert not addon.has_complete_metadata(has_listed_versions=True)
 
 
+class TestShouldRedirectToSubmitFlow(TestCase):
+    fixtures = ['base/addon_3615']
+
+    def test_no_versions_doesnt_redirect(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.should_redirect_to_submit_flow()
+
+        # Now break addon.
+        delete_translation(addon, 'summary')
+        addon = Addon.objects.get(id=3615)
+        assert not addon.has_complete_metadata()
+        addon.update(status=amo.STATUS_NULL)
+        assert addon.should_redirect_to_submit_flow()
+
+        for ver in addon.versions.all():
+            ver.delete()
+        assert not addon.should_redirect_to_submit_flow()
+
+    def test_disabled_versions_doesnt_redirect(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.should_redirect_to_submit_flow()
+
+        # Now break addon.
+        delete_translation(addon, 'summary')
+        addon = Addon.objects.get(id=3615)
+        assert not addon.has_complete_metadata()
+        addon.update(status=amo.STATUS_NULL)
+        assert addon.should_redirect_to_submit_flow()
+
+        for ver in addon.versions.all():
+            for file_ in ver.all_files:
+                file_.update(status=amo.STATUS_DISABLED)
+        assert not addon.should_redirect_to_submit_flow()
+
+    def test_only_null_redirects(self):
+        addon = Addon.objects.get(id=3615)
+        assert not addon.should_redirect_to_submit_flow()
+
+        # Now break addon.
+        delete_translation(addon, 'summary')
+        addon = Addon.objects.get(id=3615)
+        assert not addon.has_complete_metadata()
+
+        for status in amo.STATUS_CHOICES_ADDON:
+            addon.update(status=status)
+            if status == amo.STATUS_NULL:
+                assert addon.should_redirect_to_submit_flow()
+            else:
+                assert not addon.should_redirect_to_submit_flow()
```

Comment:
I personally would love to un-loop this code and rather have 5 addon.update() statements with their explicit assert statements than a for loop that can break in so many ways (or, even worse, won't break) if things change (e.g amo.STATUS_NULL get's removed from STATUS_CHOICES_ADDON)